app/services/UIBuilder.scala

```diff
@@ -138,4 +139,42 @@ class UIBuilder {
       case x :: xs => partitionComponents(xs, errors, x +: others)
     }
 
+  private def fromSequenceWithLeadingCallout(sg: StackedGroup, calloutType: CalloutType, formData: Option[FormData])
+                              (implicit stanzaIdToUrlMap: Map[String, String]): Seq[UIComponent] = {
+
+    val (callouts, visualStanzas) = extractLeadingCallouts(calloutType, sg.group, Nil)
+
+    val groupSize: Int = sg.group.size
+
+    callouts.size match {
+      case 1 =>  fromStanzas(callouts.head +: stackStanzas(visualStanzas, Nil), Nil, formData)
+      case `groupSize` =>  Seq(fromYourCallGroup(callouts))
+      case _ => fromStanzas(StackedGroup(callouts) +: stackStanzas(visualStanzas, Nil), Nil, formData)
+    }
+
+  }
+
+  @tailrec
+  private def extractLeadingCallouts(calloutType: CalloutType,
```

Comment:
this is equiv to
    val callouts: Seq[Callout] = sg.group.collect{case c: Callout if c.noteType == YourCall => c}
    val visualStanzas = sg.group.drop(callouts.length)

Reply:
Applied in calling function.